tests/std/tests/VSO_0000000_vector_algorithms/test.cpp

```diff
@@ -601,6 +602,34 @@ void test_min_max_element_special_cases() {
            == v.begin() + 2 * block_size_in_elements + last_vector_first_elem + 9);
 }
 
+template <class T>
+void test_is_sorted_until(mt19937_64& gen) {
+    using Limits = numeric_limits<T>;
+
+    uniform_int_distribution<conditional_t<sizeof(T) == 1, int, T>> dis(Limits::min(), Limits::max());
+
+    vector<T> original_input;
+    vector<T> input;
+    original_input.reserve(dataCount);
+    input.reserve(dataCount);
+
+    test_case_is_sorted_until(input, less<>{});
+    test_case_is_sorted_until(input, greater<>{});
+
+    for (size_t attempts = 0; attempts < dataCount; ++attempts) {
+        original_input.push_back(static_cast<T>(dis(gen)));
+        input = original_input;
+
+        auto it = input.begin() + uniform_int_distribution<ptrdiff_t>{0, static_cast<ptrdiff_t>(input.size() - 1)}(gen);

```

Comment:
This would be clearer if the distribution were extracted into a named variable. It's worth a line.